common/app/controllers/EmailSignupController.scala

```diff
@@ -108,37 +128,74 @@ class EmailSignupController(
   def renderFormFromName(emailType: String, listName: String): Action[AnyContent] =
     csrfAddToken {
       Action { implicit request =>
-        val id = EmailNewsletter.fromIdentityName(listName).map(_.listIdV1)
-        id match {
-          case Some(listId) =>
-            Cached(1.day)(RevalidatableResult.Ok(views.html.emailFragment(emailLandingPage, emailType, listName)))
+        val identityNewsletter = EmailNewsletter.fromIdentityName(listName)
+        identityNewsletter match {
+          case Some(newsletter) =>
+            Cached(1.day)(
+              RevalidatableResult.Ok(
+                views.html.emailFragment(
+                  emailLandingPage,
+                  emailType,
+                  newsletter.identityName,
+                  newsletter.emailEmbed,
+                ),
+              ),
+            )
           case _ => Cached(15.minute)(WithoutRevalidationResult(NoContent))
         }
       }
     }
 
-  def subscriptionResult(result: String): Action[AnyContent] =
+  def subscriptionResult(result: String, listName: String): Action[AnyContent] =
     Action { implicit request =>
-      Cached(7.days)(result match {
-        case "success" => RevalidatableResult.Ok(views.html.emailSubscriptionResult(emailLandingPage, Subscribed))
-        case "invalid" => RevalidatableResult.Ok(views.html.emailSubscriptionResult(emailLandingPage, InvalidEmail))
-        case "error"   => RevalidatableResult.Ok(views.html.emailSubscriptionResult(emailLandingPage, OtherError))
-        case _         => WithoutRevalidationResult(NotFound)
-      })
-
+      listName match {
+        case "footer" => {
+          Cached(7.days)(result match {
+            case "success" =>
+              RevalidatableResult.Ok(views.html.emailSubscriptionResultFooter(emailLandingPage, Subscribed))
+            case "invalid" =>
+              RevalidatableResult.Ok(views.html.emailSubscriptionResultFooter(emailLandingPage, InvalidEmail))
+            case "error" =>
+              RevalidatableResult.Ok(views.html.emailSubscriptionResultFooter(emailLandingPage, OtherError))
+            case _ => WithoutRevalidationResult(NotFound)
+          })
+        }
+        case listName: String => {
```

Comment:
the difference between the logic in this case and the `footer` case is mostly different due to the template used, so maybe we could pull that out as a variable and simplify this?

(pseudocode) `val template = if(footer) views.footerTemplate else views.otherTemplate`

Reply:
I think that [Alex's comment](https://github.com/guardian/frontend/pull/23051/files/288c0b3d682749459e773c36f5df8eee198dc7fe#r500907806) probably highlights the best way to deal with this. It probably makes more sense to pull this out into its own piece of logic rather than having one function handle both cases.